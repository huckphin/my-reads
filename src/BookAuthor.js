import React, { Component } from 'react';

class BookAuthor extends Component {
    render () {
        const authors = this.props.authors !== undefined ? this.props.authors.join(', ') : 'Unknown autor';

        return (
            <div className="book-authors">
                {authors}
            </div>
        )
    }
}

export default BookAuthor