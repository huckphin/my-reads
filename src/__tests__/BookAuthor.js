import React from 'react';
import BookAuthor from '../BookAuthor';
import renderer from 'react-test-renderer';

test('render a BookAuthor', () => {
    const component = renderer.create(
        <BookAuthor authors={ ["Chuck Palahniuk"] } />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});