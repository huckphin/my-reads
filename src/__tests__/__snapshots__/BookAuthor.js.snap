// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a BookAuthor 1`] = `
<div
  className="book-authors"
>
  Chuck Palahniuk
</div>
`;

exports[`render a BookAuthor with no authors 1`] = `
<div
  className="book-authors"
>
  Unknown autor
</div>
`;
